File: Modules/SEDO/EMS/SIE.Web.EMS/DevicePurs/Commands/SelUseDepaCommand.js
SIE.defineCommand('SIE.Web.EMS.DevicePurs.Commands.SelUseDepaCommand', {
    extend: 'SIE.Web.EMS.Common.Commands.SelDeviceDepaCommand',
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    /**
     * 是否可执行
     * @param {any} view
     */
    canExecute: function (view) {
        var entity = view.getParent().getCurrent();
        if (entity != null && entity.data) {
            return entity.data.INV_ORG_ID !== null;
        }
        return false;
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/DevicePurs/Commands/SelUseDepaCommand.cs
using SIE.Domain;
using SIE.EMS.Common.Entity;
using SIE.EMS.DevicePurs;
using SIE.Web.Command;
using System.Collections.Generic;

namespace SIE.Web.EMS.DevicePurs.Commands
{
    /// <summary>
    /// 添加使用部门
    /// </summary>
    [JsCommand("SIE.Web.EMS.DevicePurs.Commands.SelUseDepaCommand")]
    public class SelUseDepaCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            List<DevicePurInfo> devicePurInfos = args.Data.ToJsonObject<List<DevicePurInfo>>();
            Check.NotNullOrEmpty(devicePurInfos, nameof(devicePurInfos));
            var savedData = new EntityList<DeviceUseDepartment>();
            foreach (var item in devicePurInfos)
            {
                var deviceDepa = new DeviceUseDepartment();
                deviceDepa.DevicePurId = item.SourceId;
                deviceDepa.EnterpriseId = item.DevicePurId;                
                savedData.Add(deviceDepa);
            }
            RF.Save(savedData);
            return true;
        }
    }
}
